sys/fido2/ctap/transport/ctap_transport.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap_transport
+ * @{
+ * @file
+ *
+ * @author      Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ * @}
+ */
+
+#include "event/timeout.h"
+
+#include "fido2/ctap/transport/ctap_transport.h"
+#include "fido2/ctap/ctap.h"
+
+#if IS_USED(MODULE_FIDO2_CTAP_HID)
+#include "usb/usbus.h"
+#include "usb/usbus/hid_io.h"
+#include "fido2/ctap/transport/hid/ctap_hid.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if IS_USED(MODULE_FIDO2_CTAP_HID)
+/**
+ * @brief CTAPHID timeout handler
+ */
+static void _ctap_hid_timeout_cb(event_t *arg);
+/**
+ * @brief CTAPHID timeout event
+ */
+static event_t _ctap_hid_timeout_event = {.handler = _ctap_hid_timeout_cb};
+
+/**
+ * @brief CTAPHID event_timeout object
+ */
+static event_timeout_t _ctap_hid_event_timeout;
+#endif
+
+/**
+ * @brief CTAP stack
+ */
+static char _ctap_stack[CTAP_STACKSIZE];
+
+/**
+ * @brief CTAP transport event queue
+ */
+static event_queue_t _queue;
+
+static void *_event_loop(void* arg)
+{
+    (void)arg;
+    int ret;
+
+    ret = fido2_ctap_init();
+
+    if (ret < 0) {
+        return NULL;
+    }
+
+    event_queue_init(&_queue);
+
+#if IS_USED(MODULE_FIDO2_CTAP_HID)
```

Comment:
Right now it is very HID specific since only HID is supported as transport binding for CTAP. Once BLE / NFC is added `ctap_transport` will be less HID specific.

Reply:
To back my point: if in future BLE/NFC would be added and the file ends up in a pile of `#if ... #else´ it might still be a valid consideration to split into multiple transport files. Anyway, for the moment I don't care too much. Just wanted do share that thought.